doc/srv6/srv6_static_config_hld.md

```diff
@@ -0,0 +1,162 @@
+- [2.2 Configuration and Managment Requirements](#22-configuration-and-management-requirements)
+- [2.3 Warm Boot Requirements](#23-warm-boot-requirements)
+- [3 Feature Design](#3-feature-design)
+- [3.1 New Table in ConfigDB](#31-new-table-in-configdb)
+- [3.2 Bgpcfgd Changes](#32-bgpcfgd-changes)
+- [3.3 YANG Model](#33-yang-model)
+- [4 Unit Test](#4-unit-test)
+- [5 References ](#5-references) 
+
+# Revision
+
+| Rev  |   Date    |           Author           | Change Description      |
+| :--: | :-------: | :------------------------: | :---------------------: |
+| 0.1  | 12/5/2024 |       Changrong Wu         |  Initial version        |
+
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| ****Term**** | ****Meaning**** |
+| -------- | ----------------------------------------- |
+| BGP  | Border Gateway Protocol |
+| SID  | Segment Identifier  |
+| SRv6 | Segment Routing IPv6  |
+| SDN | Software Defined Network |
+| uSID | Micro Segment |
+| VRF  | Virtual Routing and Forwarding  |
+
+# About this Manual
+
+This document provides general information about the design of the enhancements in SONiC to support static configuration of Segmentation Routing over IPv6 protocol, which is crucial for SRv6 SDN deployment (without usage of BGP).
+
+# 1 Introuduction and Scope
+
+This document describes the high-level design of the new features in SONiC to support SRv6 SDN.
+The new features include the addtion of a new table in CONFIG_DB to allow static configuration of SRv6 and the enhancement of bgpcfgd to program FRR with input from CONFIG_DB.
+Besides, this document also define new YANG model specification and unit-test cases used to validate the aforementioned features.
+
+
+# 2 Feature Requirements
+
+## 2.1 Functional Requirements
+
+Provide ability to statically configure SRv6 SIDs for block IDs, locators and local functions from CONFIG_DB.
+
+## 2.2 Configuration and Management Requirements
+
+1. User should be able to statically configure block length, locator length and function length for SRv6.
+
+2. User should be able to statically configure a number of SIDs/uSIDs for the local functions of the switch.
+
+## 2.3 Warm Boot Requirements
+
+Warm reboot is intended to be supported for planned system warm reboot.
+
+ 
+
+# 3 Feature Design
+
+At the time of writing this document, FRR has been able to program the SRv6 related tables in APPL_DB through fpmsyncd.
+However, there is still one gap preventing SONiC being utilized for SRv6 SDN deployment.
+Specifically, there is no mechamism in SONiC allowing SDN controllers or users to directly add configuration for SRv6 without involving BGP.
+
+In this document, we first define a new **SRV6_MY_SID_TABLE** table in CONFIG_DB that serves as the configuration source of SRv6 in SONiC.
+Then, we design a new SRv6 Manager module in bgpcfgd to subscribe to the **SRV6_MY_SID_TABLE** table and compile changes in CONFIG_DB to changes in the configurations of FRR.
```

Comment:
Need to mention about the sonic specific patch to support action field in FRR as by default FRR doesn't suppor it.

Reply:
I added a note